### PR TITLE
future/settings: Migrate resultsdb optional plugin to use the new API

### PR DESCRIPTION
**Pull request by beraldoleal**
As part of the effort to remove default values scattered around the
code, this change only migrates the resultsdb optional plugin options to
use the new module. This will associate --resultsdb-api,
--resultsdb-logs, and --resultsdb-note-limit to
plugins.resultsdb.api_url, plugins.resultsdb.logs_url and
plugins.resultsdb.note_size_limit respectively on the configuration file.

### CODE REVIEW

File: optional_plugins/resultsdb/avocado_resultsdb/__init__.py [hunk 1]
```diff
@@ -156,11 +151,13 @@ class ResultsdbResult(Result):
...
-                job.config.get('stdout_claimed_by', None) is None):
+        resultsdb_logs = job.config.get('plugins.resultsdb.logs_url')
+        stdout_claimed_by = job.config.get('stdout_claimed_by')
+        if (resultsdb_logs is not None and stdout_claimed_by is None):
             log_msg = "JOB URL    : %s/%s"
-            LOG_UI.info(log_msg, job.config.get('resultsdb_logs',
-                                                os.path.basename(job.logdir)))
+            LOG_UI.info(log_msg,
+                        resultsdb_logs,
+                        os.path.basename(job.logdir))
```
Comment (willianrampazzo) [inline optional_plugins/resultsdb/avocado_resultsdb/__init__.py:160]: This seems to be a code refactor, maybe it deserves its own commit.

Comment (beraldoleal) [reply optional_plugins/resultsdb/avocado_resultsdb/__init__.py:160]: What part? I'm noot sure, just a small refactoring to fit in 80 columns using the new logic.

Comment (willianrampazzo) [reply optional_plugins/resultsdb/avocado_resultsdb/__init__.py:160]: Right, this is not related to this commit message. That is why I suggested to move it to another commit.

Comment (willianrampazzo) [reply optional_plugins/resultsdb/avocado_resultsdb/__init__.py:160]: Anyway, it does not have an impact on the module, so no problem keeping it this way.
